conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,402 @@
+}
+
+Test-Case: output
+Description: Test the type of the values in the iteration sequence of a string value is string.
+Label: foreach-stmt, string
+
+public function main() {
+    string stringVar = "abc";
+    foreach string char in stringVar {
+        io:println(char); // @output a
+                          // @output b
+                          // @output c
+    }
+}
+
+Test-Case: output
+Description: Test complition value of string value itteration is nil.
+Label: foreach-stmt, string, string:iterator, value:toBalString, while-stmt
+
+public function main() {
+    string stringVar = "a";
+    any char = "";
+    var iterator = stringVar.iterator();
+    while char != () {
+        char = iterator.next();
+        io:println(char.toBalString()); // @output {"value":"a"}
+                                        // @output ()
+    }
+
+    string stringVar2 = "";
+    io:println(stringVar2.iterator().next().toBalString()); // @output ()
+}
+
+Test-Case: error
+Description: Test the type of the values in the iteration sequence of a string value is string.
+Label: foreach-stmt, string
+
+public function main() {
+    string stringVar = "abc";
+    foreach int val in stringVar { // @error values in the iteration sequence of a string value is string
+    }
+}
+
+Test-Case: output
+Description: Test concatination of strings that will result in a value looks like numeric escape.
+Label: additive-expr, string
+
+public function main() {
+    string a = "\\" + "u{61}";
+    io:println(a); // @output \u{61}
+}
+
+Test-Case: error
+Description: Test type missmatch when assigning string literal to variables of type T, where string is not a sub type of T.
+Label: string
+
+public function main() {
+    int a = "String"; // @error string literal cannot be assigned to a variable of int
+
+    xml b = ""; // @error string literal cannot be assigned to a variable of xml
+
+    () c = ""; // @error string literal cannot be assigned to a variable of nil
+
+    byte d = "c"; // @error string literal cannot be assigned to a variable of byte
+
+    float e = "44.5"; // @error string literal cannot be assigned to a variable of float
+}
+
+Test-Case: output
+Description: Test string literal in argument context.
+Label: string
+
+public function main() {
+    printStringLiteral("Spec Tests"); // @output Spec Tests
+
+    printStringLiteral(")(*&^%$#><:~~!`"); // @output)(*&^%$#><:~~!`
+
+    printStringLiteral("12345\t\u{0039}"); // @output 12345   9
+
+    printStringLiteral("This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the \"Basic Latin\" block which goes from 0x0 to 0x7F."); // @output This code point first appeared in version 1.1 of the Unicode® Standard and belongs to the "Basic Latin" block which goes from 0x0 to 0x7F.
+}
+
+function printStringLiteral(string str) {
+    io:println(str);
+}
+
+Test-Case: output
+Description: Test string literal in default parameter context.
+Label: string
```

Comment:
We need to include relevant labels. For example, if an implementation does not support defaultable params, there needs to be a way to skip this test.

Reply:
+1 lets add `defaultable-param` to sheet as well